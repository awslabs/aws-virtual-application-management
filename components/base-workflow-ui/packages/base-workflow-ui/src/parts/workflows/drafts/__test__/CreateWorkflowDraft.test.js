/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import CreateWorkflowDraft from '../CreateWorkflowDraft';

jest.mock('../../../workflow-common/ProgressPlaceholder');

describe('CreateWorkflowDraft', () => {
  const workflowsStore = { ready: 'store-ready', load: jest.fn(), startHeartbeat: jest.fn() };
  const workflowDraftsStore = { ready: 'store-ready', load: jest.fn(), startHeartbeat: jest.fn() };
  const workflowTemplatesStore = { ready: 'store-ready', load: jest.fn(), startHeartbeat: jest.fn() };
  const onCancel = jest.fn();
  const className = 'test-className';
  it('CreateWorkflowDraft should renders correctly', () => {
    expect(
      renderer
        .create(
          <Provider
            workflowsStore={workflowsStore}
            workflowDraftsStore={workflowDraftsStore}
            workflowTemplatesStore={workflowTemplatesStore}
          >
            <CreateWorkflowDraft.WrappedComponent
              workflowsStore={workflowsStore}
              workflowDraftsStore={workflowDraftsStore}
              workflowTemplatesStore={workflowTemplatesStore}
              onCancel={onCancel}
              className={className}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
