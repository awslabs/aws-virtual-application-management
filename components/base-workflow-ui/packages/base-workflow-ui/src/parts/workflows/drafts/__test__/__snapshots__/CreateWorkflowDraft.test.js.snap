// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWorkflowDraft CreateWorkflowDraft should renders correctly 1`] = `
<div
  className="ui clearing segment p3"
>
  <form
    className="ui fluid form"
    onSubmit={[Function]}
  >
    <div
      className="ui inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted text loader"
        >
          Processing
        </div>
      </div>
    </div>
    <div
      className="mb4"
    >
      <h3
        className="ui header field mt0 mb1"
      >
        <label
          className="color-grey"
          htmlFor="draftFor--1"
        >
          Draft For
        </label>
      </h3>
      <div
        aria-disabled={false}
        aria-expanded={false}
        aria-multiselectable={false}
        className="ui fluid selection dropdown field"
        id="draftFor--1"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          className="default text"
          role="alert"
        >
          Select one
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <div
              className="text"
            >
              <div
                className="ui teal mini horizontal label"
                onClick={[Function]}
              >
                Existing
              </div>
               
              An existing workflow
            </div>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <div
              className="text"
            >
              <div
                className="ui blue mini horizontal label"
                onClick={[Function]}
              >
                New
              </div>
               
              An new workflow
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt3"
    >
      <button
        className="ui blue icon right floated button ml2"
        onClick={[Function]}
        type="submit"
      >
        Create Draft
      </button>
      <button
        className="ui right floated button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
